Code/Tools/AssetProcessor/native/unittests/AssetScannerUnitTests.cpp

```diff
@@ -7,130 +7,145 @@
-    {
-        for (AssetFileInfo foundFolder : folderList)
+        // but we're going to not watch subfolder3 recursively, so... remove these files (even though they're on disk)
+        // if this causes a failure it means that its ignoring the "do not recurse" flag:
+        expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/basefile.txt"));
+        expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/bbb/basefile.txt"));
+        expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/bbb/ccc/basefile.txt"));
+
+        QSet<QString> expectedFolders;
+        expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa");
+        expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb");
+        expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc");
+        expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd");
+        expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd/eee.fff.ggg");
+
+        PlatformConfiguration config;
+        AZStd::vector<AssetBuilderSDK::PlatformInfo> platforms;
+        config.PopulatePlatformsForScanFolder(platforms);
+        // PATH, DisplayName, PortKey, outputfolder, root, recurse, platforms
+        config.AddScanFolder(
+            ScanFolderInfo(tempPath.absolutePath(), "temp", "ap1", true, false, platforms)); // note:  "Recurse" set to false.
+        config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder1"), "", "ap2", false, true, platforms));
+        config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder2"), "", "ap3", false, true, platforms));
+        config.AddScanFolder(
+            ScanFolderInfo(tempPath.filePath("subfolder3"), "", "ap4", false, false, platforms)); // note:  "Recurse" set to false.
+        AssetScanner scanner(&config);
+
+        QList<AssetFileInfo> filesFound;
+        QList<AssetFileInfo> foldersFound;
+
+        bool doneScan = false;
+
+        connect(
+            &scanner, &AssetScanner::FilesFound, this,
+            [&filesFound](QSet<AssetFileInfo> fileList)
+            {
+                for (AssetFileInfo foundFile : fileList)
+                {
+                    filesFound.push_back(foundFile);
+                }
+            });
+
+        connect(
+            &scanner, &AssetScanner::FoldersFound, this,
+            [&foldersFound](QSet<AssetFileInfo> folderList)
+            {
+                for (AssetFileInfo foundFolder : folderList)
+                {
+                    foldersFound.push_back(foundFolder);
+                }
+            });
+
+        connect(
+            &scanner, &AssetScanner::AssetScanningStatusChanged, this,
+            [&doneScan](AssetProcessor::AssetScanningStatus status)
+            {
+                if ((status == AssetProcessor::AssetScanningStatus::Completed) || (status == AssetProcessor::AssetScanningStatus::Stopped))
+                {
+                    doneScan = true;
```

Comment:
just a suggestion, but it might be good to encapsulate this "is scan done" into a single API like "scanner.IsDone()" so that clients do not have to check for these specific flags

nothing that blocks this check in, just a suggestion

Reply:
I just ported what was already there, I didn't change any of the logic of the test.